src/core/connection.h

```diff
@@ -461,6 +461,12 @@ typedef struct QUIC_CONNECTION {
     //
     QUIC_CID* OrigDestCID;
 
+    //
+    // An app configured prefix for all connection IDs.
+    //
+    uint8_t CidPrefixLength;
+    uint8_t* CidPrefix;
```

Comment:
An inline struct here storing the length and the 6 prefix bytes would actually have a smaller memory footprint here then the separate allocation, in addition to being cleaner because of no need to maintain the separate allocation. You could actually store 15 bytes of inline data in the amount of space this takes up.

Reply:
Why does this have 15 bytes of padding instead of 7? 7 bytes of padding would make the pointer aligned on an 8-byte boundary (on 64-bit), and the following array is aligned after the pointer.

Reply:
It has 7 bytes of padding, but then you have 8 bytes for the pointer itself. Thats where the 15 comes from. Its 15 bytes of data, not 15 bytes of padding.

Reply:
On 32 bit, you have 4 bytes of data and 3 bytes of padding. So you could even fit the 7 bits of data needed here in 32 bit as well. So its a win win in all cases.